### PR TITLE
Delete VM with admin creentials

### PR DESCRIPTION
**Pull request by hmunfru**
https://jirapdi.tid.es/browse/CLAUDIA-4990
@jframos

### CODE REVIEW

**Comment by flopezag** [comment]
Where is the Unit Test of this change?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 1]
```diff
@@ -605,24 +606,32 @@ public String deployVM(String org, String vdc, String service, String vmName, Pa
...
+     * @param tierInstance
+     * @throws InfrastructureException
+     */
     public void undeployVMReplica(ClaudiaData claudiaData, TierInstance tierInstance) throws InfrastructureException {
         log.debug("Undeploy VM replica " + tierInstance.getName() + " for region " + tierInstance.getTier().getRegion()
-                + " and user " + tierInstance.getTier().getVdc());
+            + " and user " + tierInstance.getTier().getVdc());
         try {
-
+            OpenStackAccess openStackAccess = openStackRegion.getTokenAdmin();
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: Something estrange here. Do you get the admin token but you do not use it? This PR is just regarding the use of the admin token in order to delete the operations? You are using openStackAccess.getToken() and not openStackRegion.getTokenAdmin()?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 2]
```diff
@@ -605,24 +606,32 @@ public String deployVM(String org, String vdc, String service, String vmName, Pa
...
             String region = tierInstance.getTier().getRegion();
-            String token = claudiaData.getUser().getToken();
-            String vdc = tierInstance.getTier().getVdc();
+            log.debug("tenantId " + openStackAccess.getTenantId());
+            log.debug("token " + openStackAccess.getToken());
 
-            openStackUtil.deleteServer(tierInstance.getVM().getVmid(), region, token, vdc);
+            openStackUtil.deleteServer(tierInstance.getVM().getVmid(), tierInstance.getTier().getRegion(),
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: tierInstance.getTier().getRegion() is not the valraible region?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 3]
```diff
@@ -605,24 +606,32 @@ public String deployVM(String org, String vdc, String service, String vmName, Pa
...
+                openStackAccess.getToken(), openStackAccess.getTenantId());
 
             checkDeleteServerTaskStatus(tierInstance, claudiaData);
             log.debug("Undeployed VM replica " + tierInstance.getName() + " for region "
-                    + tierInstance.getTier().getRegion() + " and user " + tierInstance.getTier().getVdc());
+                + tierInstance.getTier().getRegion() + " and user " + tierInstance.getTier().getVdc());
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: tierInstance.getTier().getRegion() is not the valraible region?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 4]
```diff
@@ -605,24 +606,32 @@ public String deployVM(String org, String vdc, String service, String vmName, Pa
...
 
             if (tierInstance.getTier().getFloatingip().equals("true")) {
                 log.debug("Delete floating ip ");
-                openStackUtil.disAllocateFloatingIP(region, token, vdc, tierInstance.getVM().getFloatingIp());
+                openStackUtil.disAllocateFloatingIP(region, openStackAccess.getToken(), openStackAccess.getTenantId(),
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: openStackAccess.getToken() and openStackAccess.getTenantId() firstly should be a variable due to you are calling them twice in your code and secondly are you sure that you take the admin token callin getToken()?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java (hunk 3 above)
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: Second question, we obtain the user from tierInstance.getTier().getVdc()?

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java [hunk 5]
```diff
@@ -184,6 +187,27 @@ public void testDeployVMEssex() throws Exception {
     }
 
     @Test
+    public void testUnDeployVMReplica() throws Exception {
+
+        TierInstance tierInstance = new TierInstance();
+        tierInstance.setTier(tier);
+        VM vm = new VM();
+        vm.setHostname("hotname");
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java]: typo not important "hotname"

File: core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java [hunk 6]
```diff
@@ -184,6 +187,27 @@ public void testDeployVMEssex() throws Exception {
...
+        tierInstance.setVM(vm);
+
+        OpenStackAccess access = new OpenStackAccess();
+        access.setTenantId("tenantId");
+        access.setToken("token");
+        when(openStackRegion.getTokenAdmin()).thenReturn(access);
+        when(openStackUtil.deleteServer(anyString(), anyString(), anyString(), anyString())).thenReturn("");
```
Comment (flopezag) [inline core/src/test/java/com/telefonica/euro_iaas/paasmanager/claudia/ClaudiaClientOpenStackImplTest.java]: Why are not closing the details of this "when" when you know with the data that you have that the data is fixed...

The same with the third "when"

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java (hunk 1 above)
Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: I need the admin token. I use this admin token (openstackaccess) in the following lines
penStackUtil.disAllocateFloatingIP(region, openStackAccess.getToken(), openStackAccess.getTenantId(),

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java (hunk 3 above)
Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: the vdc is the tenant id of the user. For admin request, we need the tenantid for the admin user

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java (hunk 1 above)
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: NTC

**Comment by flopezag** [comment]
LGTM

**Pull request merged**